Xamarin.Forms.Platform.Android/Renderers/ShellSearchView.cs

```diff
@@ -209,6 +209,7 @@ protected virtual void LoadView(SearchHandler searchHandler)
 			_textBlock.Adapter = new ShellSearchViewAdapter(SearchHandler, _shellContext);
 			_textBlock.ItemClick += OnTextBlockItemClicked;
 			_textBlock.SetDropDownBackgroundDrawable(new ClipDrawableWrapper(_textBlock.DropDownBackground));
+			_textBlock.FocusChange += (sender, eventArgs) => { OnSearchHandlerFocusChange(sender, eventArgs, searchHandler); };
```

Comment:
isn't better to use the same pattern as the `ItemClick` event and unsubscribe it on `Dispose` method?

Reply:
Hi @pictos , at first I tried to register the OnSearchHandlerFocusChange following the same pattern than `_textBlock.ItemClick += OnTextBlockItemClicked` but the difference here is that `_textBlock` is a class scoped variable whereas the `searchHandler` is a function scoped var (passed through params into `LoadView(SearchHandler searchHandler)`.

Since I needed (actually, wanted) to simply invoke the `SearchHandler.SetValue` instance scope function I needed to pass through the `searchHandler`  variable to the `OnSearchHandlerFocusChange`.
This caused the number of arguments for `_textBlock.FocusChange` and `OnSearchHandlerFocusChange` to differ, therefore impeding the use of the aforementioned `ItemClick`  pattern.

So, having this in mind, it's true that a 'shortened' assignation could also have been
`_textBlock.FocusChange += (sender, e) => searchHandler.SetValue(SearchHandler.IsFocusedPropertyKey, e.HasFocus)` where everything falls in a same line, skipping the creation of the `OnSearchHandlerFocusChange` function but I think this option may be less explicit in terms of "Code readability".

Now, having an even deeper look into the code, I've just realised that `LoadView(SearchHandler searcHandler)` is passing through its `searchHandler` param the `SearchHandler` that's actually accessible at an instance level scope, as it's an automatic property declared within the class :')

So it's true, I guess your review suggestion is more than appropiate, and that I can actually go for it. Sorry for the overextended reasoning. Be right back.